File: 011/011.py
class Solution:
    def maxArea(self, height):

        ## Input:
        # height: List[int]
        ## Output:
        # int

        i = 0
        j = len(height) - 1
        max_area = 0

        while (i < j):
            if (height[i] <= height[j]):
                area = height[i] * (j - i)
                i += 1
            else:
                area = height[j] * (j - i)
                j -= 1
            if (area > max_area):
                max_area = area
        
        return max_area

File: 011/011.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int i = 0;
        int j = height.size() - 1;
        int max_area = 0;
        int area;

        while (i < j)
        {
            if (height[i] <= height[j])
            {
                area = height[i] * (j - i);
                i++;
            }
            else
            {
                area = height[j] * (j - i);
                j--;
            }

            if (area > max_area)
            {
                max_area = area;
            }
        }

        return max_area;
    }
};